modelling/src/test/java/org/axonframework/modelling/saga/repository/CachingSagaStoreTest.java

```diff
@@ -219,4 +224,43 @@ StubSaga.class, sagaId, mock(StubSaga.class), associationValues
             fail("An unexpected exception occurred during concurrent invocations on the CachingSagaStore.", e);
         }
     }
+
+    @Test
+    void canHandleConcurrentReadsAndWritesThroughAnnotatedSagaRepository() {
+        SagaRepository<StubSaga> sagaRepository = AnnotatedSagaRepository.<StubSaga>builder()
+                                                                         .sagaType(StubSaga.class)
+                                                                         .sagaStore(testSubject)
+                                                                         .build();
+        int concurrentOperations = 64;
+        ExecutorService executor = Executors.newFixedThreadPool(16);
```

Comment:
It couldn't hurt to add a finally clause shutting down the executor.

Reply:
Smart! I'll do it for the other concurrency test too :+1: